File: wiz-event-mall-java/src/main/java/com/example/wizeventmalljava/api/event/dto/EventInfoResponse.java
package com.example.wizeventmalljava.api.event.dto;

import java.time.LocalDateTime;

import com.example.wizeventmalljava.domain.event.entity.Event;
import com.example.wizeventmalljava.domain.event.entity.EventStatus;

public record EventInfoResponse(
	Long eventId,
	String eventName,
	LocalDateTime startAt,
	Long winnerCount,
	EventStatus status
) {
	public static EventInfoResponse from(Event event) {
		return new EventInfoResponse(
			event.getId(),
			event.getName(),
			event.getStartAt(),
			event.getWinnerCount(),
			event.getStatus()
		);
	}
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/api/event/dto/EventInfoResponse.kt
package com.example.wizeventmall.api.event.dto

import com.example.wizeventmall.domain.event.entity.Event
import com.example.wizeventmall.domain.event.entity.EventStatus
import java.time.LocalDateTime

data class EventInfoResponse(
    val eventId: Long,
    val eventName: String,
    val startAt: LocalDateTime,
    val winnerCount: Long,
    val status: EventStatus
) {
    companion object {
        fun from(event: Event) = EventInfoResponse(
            eventId = event.id!!,
            eventName = event.name,
            startAt = event.startAt,
            winnerCount = event.winnerCount,
            status = event.status
        )
    }
}